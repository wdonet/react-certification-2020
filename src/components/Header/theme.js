const base = {
  easeOutBack: 'cubic-bezier(0.34, 1.56, 0.64, 1)',
  colorWhite: 'rgb(255, 255, 255)',
  colorBlack: 'rgb(0, 0, 0)',
};

const dark = {
  id: 'dark',
  ...base,
  backgroundColor: '#333',
  textColor: 'black',
};

const light = {
  id: 'light',
  ...base,
  backgroundColor: '#666',
  textColor: 'white',
};

export const theme = { dark, light };
