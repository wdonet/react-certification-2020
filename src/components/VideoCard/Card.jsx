import React from 'react';
import Card from 'react-bootstrap/Card';

function VideoCard({ src, title, description }) {
  return (
    <Card
      style={{ width: '18rem', height: '30rem', marginBottom: '1rem', marginTop: '1rem' }}
    >
      <Card.Body>
        <Card.Img variant="top" src={src} />
        <Card.Title>{title}</Card.Title>
        <Card.Text>{description}</Card.Text>
      </Card.Body>
    </Card>
  );
}
export default VideoCard;
