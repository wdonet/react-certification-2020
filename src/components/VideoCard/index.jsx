import React from 'react';

import {
  CardContainer,
  VideoDetails,
  Title,
  Description,
  VideoImg,
  TitleContainer,
  ChannelLogo,
} from './styled';

function getMonthsTillToday(date) {
  let months;
  const today = new Date();
  months = (today.getFullYear() - date.getFullYear()) * 12;
  months -= today.getMonth();
  months += date.getMonth();
  return months <= 0 ? 0 : months;
}

const VideoCard = ({ title, channelTitle, publishedAt, image }) => {
  const months = getMonthsTillToday(new Date(publishedAt));

  return (
    <CardContainer>
      <VideoImg src={image} />
      <VideoDetails>
        <TitleContainer>
          <ChannelLogo />
          <Title>{title}</Title>
        </TitleContainer>
        <Description>
          {channelTitle}
          <br /> {months} months ago
        </Description>
      </VideoDetails>
    </CardContainer>
  );
};

export default VideoCard;
