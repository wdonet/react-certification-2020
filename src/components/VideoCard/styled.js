import styled from 'styled-components';

const CardContainer = styled.li`
  background-color: ${({ theme }) => theme.background.secondary};
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  overflow: hidden;

  :hover {
    transform: scale(1.1);
  }
`;

const VideoImg = styled.img`
  width: 100%;
`;

const VideoDetails = styled.div`
  padding: 0.5em;
`;

const TitleContainer = styled.div`
  display: flex;
  flex-direction: row;
`;

const ChannelLogo = styled.img``;

const Title = styled.p`
  margin: 0px;
  font-size: 0.9em;
  color: ${({ theme }) => theme.font.primary};
  font-weight: 600;
  text-align: left;
`;

const Description = styled.p`
  font-size: 0.8em;
  text-align: left;
  color: ${({ theme }) => theme.font.secondary};
  margin: 8px;
  font-weight: 500;
`;

export {
  CardContainer,
  VideoDetails,
  Title,
  Description,
  VideoImg,
  TitleContainer,
  ChannelLogo,
};
