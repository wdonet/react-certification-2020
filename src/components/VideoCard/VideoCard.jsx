import React from 'react';
import { Container, Title } from './VideoCard.styles';

const VideoCard = ({ video, handleVideoSelected }) => {
  return (
    <Container>
      <div role="button">
        <img alt={video.snippet.description} src={video.snippet.thumbnails.default.url} />
        <div>
          <Title>{video.snippet.title}</Title>
        </div>
      </div>
      <button type="button" onClick={() => handleVideoSelected(video)}>
        View details
      </button>
    </Container>
  );
};

export default VideoCard;
