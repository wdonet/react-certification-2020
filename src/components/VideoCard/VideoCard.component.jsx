import React from 'react';
import { decode } from 'html-entities';
import { Link } from 'react-router-dom';
import { VideoCardContainer, Details, ImageContainer } from './VideoCard.styled';

function VideoCard({ image, title, description, videoId, noDescription = false }) {
  return (
    <VideoCardContainer noDescription={noDescription}>
      <Link to={`/video/${videoId}`}>
        <ImageContainer
          role="img"
          aria-label="thumbnail"
          img={image}
          noDescription={noDescription}
        />
        <Details noDescription={noDescription}>
          <h2>{decode(title)}</h2>
          {noDescription || <p>{decode(description)}</p>}
        </Details>
      </Link>
    </VideoCardContainer>
  );
}

export default VideoCard;
