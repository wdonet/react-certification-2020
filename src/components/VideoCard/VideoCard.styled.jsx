import styled from 'styled-components';

export const VideoCardContainer = styled.div`
  width: ${({ noDescription }) => (noDescription ? '14em' : '20em')};
  height: ${({ noDescription }) => (noDescription ? '10em' : '20em')};
  margin: 0.5em;
  overflow: hidden;
  background-color: ${({ theme }) => theme.videoCardBackground};
  border: 1px solid rgba(0, 0, 0, 0.15);
  border-radius: 2px;
  color: ${({ theme }) => theme.videoCardTextColor};
`;

export const ImageContainer = styled.div`
  width: 100%;
  height: ${({ noDescription }) => (noDescription ? '60%' : '40%')};
  background-image: url(${({ img }) => img});
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
`;

export const Details = styled.div`
  padding: 1em;

  & h2 {
    font-size: ${({ noDescription }) => (noDescription ? '0.6rem' : '1rem')};
    text-align: ${({ noDescription }) => (noDescription ? 'center' : 'left')};
    font-weight: 800;
    margin: 0 0 0.35em 0;
  }

  & p {
    font-size: 0.785rem;
    font-weight: 400;
    margin: 0;
  }
`;
