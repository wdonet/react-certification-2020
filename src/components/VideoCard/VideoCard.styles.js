import styled from 'styled-components';

export const Container = styled.div`
  width: 350px;
  border: 1px solid #cccccc;
  border-radius: 5px;
  padding: 20px;
  background-color: #e7e7fc;
`;
