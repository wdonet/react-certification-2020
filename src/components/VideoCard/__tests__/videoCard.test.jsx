import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

import VideoCard from '../index';

import { lightTheme } from '../../../theme';

describe('VideoCard', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={lightTheme}>
          <VideoCard
            title="Title of the video"
            channelTitle="Wizeline"
            publishedAt={1232143242}
            image="thisistheurl.com"
          />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
