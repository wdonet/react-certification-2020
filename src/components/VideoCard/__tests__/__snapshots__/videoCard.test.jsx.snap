// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard renders correctly 1`] = `
.c0 {
  background-color: #FFFFFF;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  overflow: hidden;
}

.c0:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c1 {
  width: 100%;
}

.c2 {
  padding: 0.5em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  margin: 0px;
  font-size: 0.9em;
  color: #333333;
  font-weight: 600;
  text-align: left;
}

.c5 {
  font-size: 0.8em;
  text-align: left;
  color: #666666;
  margin: 8px;
  font-weight: 500;
}

<li
  className="c0"
>
  <img
    className="c1"
    src="thisistheurl.com"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <img
        className=""
      />
      <p
        className="c4"
      >
        Title of the video
      </p>
    </div>
    <p
      className="c5"
    >
      Wizeline
      <br />
       
      609
       months ago
    </p>
  </div>
</li>
`;
